lighthouse-core/config/config.js

```diff
@@ -401,18 +402,21 @@ class Config {
   }
 
   /**
-   * @param {LH.Config.SettingsJson=} settings
+   * @param {LH.Config.SettingsJson=} settingsJson
    * @param {LH.Flags=} flags
    * @return {LH.Config.Settings}
    */
-  static initSettings(settings = {}, flags) {
+  static initSettings(settingsJson = {}, flags) {
     // Fill in missing settings with defaults
     const {defaultSettings} = constants;
-    const settingWithDefaults = merge(deepClone(defaultSettings), settings, true);
+    const settingWithDefaults = merge(deepClone(defaultSettings), settingsJson, true);
 
     // Override any applicable settings with CLI flags
     const settingsWithFlags = merge(settingWithDefaults || {}, cleanFlagsForSettings(flags), true);
 
+    // Use locale specified in flags or settings, allowing i18n system to select fallback if needed.
+    settingsWithFlags.locale = i18n.lookupLocale((flags && flags.locale) || settingsJson.locale);
```

Comment:
can you extract `(flags && flags.locale) || settingsJson.locale` to a const that is defined above L415. between fallbacks and merges i want to be quite clear what is affecting locale and what isnt. (in this case the merge apparently doesnt)

Reply:
Hmm, I was putting it down here to *attempt* to make it clear that you can just ignore everything above for `locale`, this is the place that it's determined and set (and that `defaultSettings.locale` can't possibly influence the value).

Do you want me to set it on `settingWithDefaults` or `cleanFlagsForSettings(flags)` and let the merge merge it in?